framework/src/problems/FEProblemBase.C

```diff
@@ -8856,8 +8856,6 @@ FEProblemBase::getFVMatsAndDependencies(
     for (std::shared_ptr<MaterialBase> face_mat : this_face_mats)
       if (face_mat->ghostable())
       {
-        mooseAssert(!face_mat->hasStatefulProperties(),
-                    "Finite volume materials do not currently support stateful properties.");
```

Comment:
we could change these two checks into something else:
- not an assert, ran for all users
- checking that no material depends on a FV variable and is stateful

it would go just 3 lines below basically